pkg/nats/transport/nats.go

```diff
@@ -70,6 +83,28 @@ func (c *NATSTransportConfig) Validate() error {
 	} else {
 		mErr = errors.Join(mErr, validate.IsNotEmpty(c.Orchestrators, "missing orchestrators"))
 	}
+
+	serverCertProvided := len(c.ServerTLSCert) > 0
+	serverKeyProvided := len(c.ServerTLSKey) > 0
+
+	if serverCertProvided != serverKeyProvided {
+		mErr = errors.Join(
+			mErr,
+			fmt.Errorf("both ServerTLSCert and ServerTLSKey must be set together"),
+		)
+	}
+
+	if serverCertProvided && serverKeyProvided && c.ServerTLSTimeout < 0 {
+		mErr = errors.Join(
+			mErr,
+			fmt.Errorf("NATS ServerTLSTimeout must be a positive number, got: %d", c.ServerTLSTimeout),
+		)
+	}
+
+	if serverCertProvided && serverKeyProvided && c.ServerTLSTimeout == 0 {
+		c.ServerTLSTimeout = NATSServerDefaultTLSTimeout
+	}
```

Comment:
I don't think this fits in `Validate` method

Reply:
I am on the fence with this, I added it here to be able to unit tests it, otherwise I needed to refactor the code a lot.

Reply:
Fixed. Logic removed from the validate.